src/Traits/CacheProxiesList.php

```diff
@@ -0,0 +1,28 @@
+<?php
+
+namespace Softonic\Proxy\Guzzle\Middleware\Traits;
+
+use Softonic\Proxy\Guzzle\Middleware\Exceptions\ProxiesNotAvailable;
+
+trait CacheProxiesList
+{
+    public function getProxyUsingCache()
+    {
+        try {
+            $item = $this->cache->getItem(self::CACHE_KEY);
```

Comment:
I think it's a little weird to use a member variable is not defined in the same scope, we assume that the code uses this trait has a `cache` member variable